pkg/apiserver/rest/dualwriter_mode3.go

```diff
@@ -42,30 +44,57 @@ func (d *DualWriterMode3) Mode() DualWriterMode {
 const mode3Str = "3"
 
 // Create overrides the behavior of the generic DualWriter and writes to LegacyStorage and Storage.
-func (d *DualWriterMode3) Create(ctx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
+func (d *DualWriterMode3) Create(ctx context.Context, in runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
 	var method = "create"
 	log := d.Log.WithValues("method", method)
+
 	ctx = klog.NewContext(ctx, log)
 
-	startStorage := time.Now()
-	created, err := d.Storage.Create(ctx, obj, createValidation, options)
+	accIn, err := meta.Accessor(in)
 	if err != nil {
+		return nil, err
+	}
+
+	if accIn.GetUID() != "" {
+		return nil, fmt.Errorf("there is an UID and it should not: %v", accIn.GetUID())
+	}
+
+	startStorage := time.Now()
+	storageObj, errObjectSt := d.Storage.Create(ctx, in, createValidation, options)
+	d.recordStorageDuration(errObjectSt != nil, mode3Str, d.resource, method, startStorage)
+	if errObjectSt != nil {
 		log.Error(err, "unable to create object in storage")
-		d.recordLegacyDuration(true, mode3Str, d.resource, method, startStorage)
-		return created, err
+		return storageObj, errObjectSt
 	}
-	d.recordStorageDuration(false, mode3Str, d.resource, method, startStorage)
 
-	go func() {
-		ctx, cancel := context.WithTimeoutCause(ctx, time.Second*10, errors.New("legacy create timeout"))
-		defer cancel()
+	createdCopy := storageObj.DeepCopyObject()
+
+	//nolint:errcheck
+	go d.createOnLegacyStorage(ctx, createdCopy, createValidation, options)
```

Comment:
what if we need to roll back to legacy? is there a reverse sync?

Reply:
there aren't plans for that, but it's actually something we want to discuss. in the team: if there is one, that will mean that mode 3 will be pretty similar to mode 2 :/